### PR TITLE
Add handling string.match(regex)

### PR DESCRIPTION
**Pull request by IzabelaSobczyk**
## What:
Add handling "string match regex"

## Why:
I need it for user data form (auction modal - for one of error after bid action we have to show this form). I have to validate text from each textInput. I need to check for example if in phone number someone typed "abc defg" instead of "121 121 121" or for name "sun123" instead of "John"

### CODE REVIEW

**Review by KTGypsy — DISMISSED** [review]

File: operations-stdlib/src/commonMain/kotlin/string/Match.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        val regex = regexPattern.toRegex(convertedOptions)
+        return if(convertedOptions.any { it == RegexOption.MULTILINE}) {
+            val splittedString = text.split("\n")
+            return splittedString.all { regex.matches(it) }
+        } else {
+            regex.matches(text)
+        }
+    }
+
+    private fun MatchArguments.invokeRegex() = if (regexOptions.isNullOrEmpty()) matchBasic(regexPattern, text)
```
Comment (KTGypsy) [inline operations-stdlib/src/commonMain/kotlin/string/Match.kt:42]: using curly brackets would make it more readable when the line is broken

File: operations-stdlib/src/commonMain/kotlin/string/Match.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+object Match : StandardLogicOperation, StringUnwrapStrategy {
+    private const val TEXT_ARG_INDEX = 0
+    private const val REGEX_PATTERN_ARG_INDEX = 1
+    private const val REGEX_OPTIONS_ARG_INDEX = 2
+    override fun evaluateLogic(expression: Any?, data: Any?): Any? =
+        expression.asList.toOperationArguments()?.invokeRegex()
+
+    private fun List<Any?>.toOperationArguments(): MatchArguments? = runCatching {
+
```
Comment (KTGypsy) [inline operations-stdlib/src/commonMain/kotlin/string/Match.kt:14]: empty line

**Review by lsolniczek — APPROVED** [review]

**Review by KTGypsy — CHANGES_REQUESTED** [review]

File: operations-stdlib/src/commonMain/kotlin/string/Match.kt [hunk 3]
```diff
@@ -0,0 +1,50 @@
+package string
+
+import operation.StandardLogicOperation
+import utils.asList
+
+object Match : StandardLogicOperation, StringUnwrapStrategy {
+    private const val TEXT_ARG_INDEX = 0
+    private const val REGEX_PATTERN_ARG_INDEX = 1
+    private const val REGEX_OPTIONS_ARG_INDEX = 2
+    override fun evaluateLogic(expression: Any?, data: Any?): Any? =
```
[the remark this answers is not in the document]
Comment (IzabelaSobczyk) [reply operations-stdlib/src/commonMain/kotlin/string/Match.kt:11]: Changes added

File: operations-stdlib/src/commonMain/kotlin/string/Match.kt (hunk 2 above)
Comment (IzabelaSobczyk) [reply operations-stdlib/src/commonMain/kotlin/string/Match.kt]: Changes added

**Review by KTGypsy — APPROVED** [review]

**Review by pavlo-davydiuk — APPROVED** [review]

**Pull request merged**